src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -800,7 +800,9 @@ func (s *service) Kill(ctx context.Context, r *taskAPI.KillRequest) (_ *ptypes.E
 			return empty, errors.New("The exec process does not exist")
 		}
 		processStatus = execs.status
-	} else {
+	}
+
+	if signum == syscall.SIGKILL {
```

Comment:
I don't think this solves https://github.com/kata-containers/kata-containers/issues/5831. If you trace the code, `r.All = true` is translated to `execID = ""` in[ kata_agent.go](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/kata_agent.go#L1647-L1649).